cmd/gateway-gcs.go

```diff
@@ -151,14 +153,62 @@ func gcsToObjectError(err error, params ...string) error {
 // gcsProjectIDRegex defines a valid gcs project id format
 var gcsProjectIDRegex = regexp.MustCompile("^[a-z][a-z0-9-]{5,29}$")
 
-// isValidGCSProjectId - checks if a given project id is valid or not.
-// Project IDs must start with a lowercase letter and can have lowercase
-// ASCII letters, digits or hyphens. Project IDs must be between 6 and 30 characters.
-// Ref: https://cloud.google.com/resource-manager/reference/rest/v1/projects#Project (projectId section)
 func isValidGCSProjectID(projectID string) bool {
+	// Checking projectID format
 	return gcsProjectIDRegex.MatchString(projectID)
 }
 
+// checkGCSProjectID - checks if a given project id is valid and exists.
+// Project IDs has a specific format described here in the following link:
+// https://cloud.google.com/resource-manager/reference/rest/v1/projects#Project (projectId section)
+// Besides, this function checks if the project ID does really exist using
+// resource manager API.
+func checkGCSProjectID(ctx context.Context, projectID string) error {
```

Comment:
> Can't this be unit testable?

only isValidGCSProjectID can be testable. Otherwise we need a real GCS account with an existing bucket.